repos/system_upgrade/el7toel8/actors/nischeck/libraries/nischeck.py

```diff
@@ -0,0 +1,152 @@
+import re
+import ipaddress
+
+from leapp import reporting
+
+
+def scan():
+    """
+    Check configuration of /etc/yp.conf and /etc/nsswitch.conf. Address of NIS server cannot be
+    specified by host name if NIS is used to resolve host names.
+    """
+
+    hostnames = hostnames_in_yp_conf()
+    data, _ = scan_nsswitch()
+
+    if "hosts" in data:
+        if "nis" in data["hosts"] and hostnames:
+            report_error(hostnames)
+
+
+def report_error(hostnames):
+    reporting.create_report([
+        reporting.Title('Unsupported NIS configuration found'),
+        reporting.Summary("NIS may be used for domain name resolution only if NIS "
+                          "server is specified by IP. NIS servers specified by "
+                          "host name: {}".format(", ".join(hostnames))),
+        reporting.Severity(reporting.Severity.MEDIUM)
+    ])
+
+
+def hostnames_in_yp_conf():
+    """
+    Scan /etc/yp.conf, return if file contains domain name
+    """
+
+    path = "/etc/yp.conf"
+    try:
+        with open(path) as f:
+            lines = f.readlines()
+    except IOError:
+        return False
+
+    # remove comments
+    lines = [l.split("#", 1)[0] for l in lines]
+
+    re_server = r"\s*domain\s+\S+\s+server\s+(\S+)\s*"
+    re_ypserver = r"\s*ypserver\s+(\S+)\s*"
+
+    # Find server name specified in form: domain NISDOMAIN server HOSTNAME
+    mtchs = [m for m in [re.match(re_server, l) for l in lines] if m]
```

Comment:
Could you please combine L50 and L53?
 And if it's not too much to ask - please use `matches` instead of `mtchs`.
```
matches = [m for m in [re.match("{re_server}|{re_ypserver}".format(re_server=re_server, re_ypsever=re_ypserver), l) for l in lines] if m]
```

Reply:
Thanks for the review, I have applied all your recommendation except this one, because it does not improve code readability from my point of view.